File: src/main/java/com/example/member/service/MemberService.java
package com.example.member.service;

import com.example.member.dto.MemberDto;
import com.example.member.entity.Member;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface MemberService {
    
    // 회원가입
    ResponseEntity<String> joinMember(MemberDto memberDto);

    // 회원 조회
    ResponseEntity<List<Member>> findMember(MemberDto memberDto);

    // 회원정보 업데이트
    ResponseEntity <String> updateMember(MemberDto memberDto);
    
    // 회원 탈퇴
    ResponseEntity<String> deleteMember(MemberDto memberDto);

}

File: src/main/kotlin/com/example/member/service/MemberService.kt
package com.example.member.service

import com.example.member.dto.MemberDto
import com.example.member.entity.Member
import org.springframework.http.ResponseEntity

interface MemberService {

    // 회원가입
    fun joinMember(memberDto: MemberDto): ResponseEntity<String>

    // 회원 조회
    fun findMember(memberDto: MemberDto): ResponseEntity<MutableList<Member>>

    // 회원정보 업데이트
    fun updateMember(memberDto: MemberDto): ResponseEntity<String>

    // 회원 탈퇴
    fun deleteMember(memberDto: MemberDto): ResponseEntity<String>

    fun doLogin(memberDto: MemberDto): ResponseEntity<String>

}
